packages/web-component/src/lib/helpers/helpers.ts

```diff
@@ -330,10 +375,21 @@ export const getChromiumVersion = (
 // As an optimization - We can show first screen if we have startScreenId and we don't have oidcIdpStateId
 // - If there startScreenId it means that the sdk can show the first screen and we don't need to wait for the sdk to return the first screen
 // - If there is a oidcIdpStateId - we can't skip this call because the sdk may
+// - If there is a samlIdpStateId - we can't skip this call because the sdk may
+// - If there is a samlIdpUsername - we can't skip this call because the sdk may
+// - If there is a ssoAppId - we can't skip this call because the sdk may
 export const showFirstScreenOnExecutionInit = (
   startScreenId: string,
-  oidcIdpStateId: string
-): boolean => startScreenId && !oidcIdpStateId;
+  oidcIdpStateId: string,
+  samlIdpStateId: string,
+  samlIdpUsername: string,
+  ssoAppId: string
+): boolean =>
+  startScreenId &&
+  !oidcIdpStateId &&
```

Comment:
Is there a scenario where only some of these params has a value?
Do we want to skip the optimization if only some has a value? or we should have it all in order for this feature to work?

Reply:
yea, for example when oidcIdpStateId has a value samlIdpStateId will not have a value (for sure as its another scenarios)

Reply:
Thanks to @tomerlichtash we make the logic here more clear as well.